File: API/Features/Sales/Prices/Dtos/PriceReadDto.cs
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.Sales.Prices {

    public class PriceReadDto : IMetadata {

        // PK
        public int Id { get; set; }
        // Fields
        public string Code { get; set; }
        public string Description { get; set; }
        public string EnglishDescription { get; set; }
        public bool IsIndividual { get; set; }
        public int Length { get; set; }
        public decimal NetAmount { get; set; }
        public decimal VatPercent { get; set; }
        public decimal VatAmount { get; set; }
        public decimal GrossAmount { get; set; }
        // Navigation
        public SimpleEntity HullType { get; set; }
        public SimpleEntity SeasonType { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}

File: FrontEnd/src/app/features/sales/prices/classes/dtos/price-read-dto.ts
import { Metadata } from 'src/app/shared/classes/metadata'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface PriceReadDto extends Metadata {

    id: number
    hullType: SimpleEntity
    periodType: SimpleEntity
    seasonType: SimpleEntity
    code: string
    description: string
    englishDescription: string
    isIndividual: boolean
    length: number
    netAmount: number
    vatPercent: number
    vatAmount: number
    grossAmount: number

}
